bin/handler-opsgenie.rb

```diff
@@ -83,6 +83,8 @@ def default_message
   end
 
   def event_id
+    return @event['check']['opsgenie']['alias'] unless (@event['check']['opsgenie'].nil? or @event['check']['opsgenie']['alias'].nil?)
```

Comment:
In ruby `or` has a lower precedence with checking as `||` so we should use this where possible.

http://phrogz.net/programmingruby/language.html#table_18.4

Also the parenthesis around unless are unnecessary. While I personally don't mind them as I think it adds more clarity to those foreign to, ruby many rubyists find it annoying as it takes up extra chars on the line.

Reply:
Done.  (Although I never found arguments about number of characters on a line, or to type, to be very compelling.  Oh well.)

Reply:
I generally agree on the number of chars on a relatively short line like this and I do think that often many people take this way too far. Much of the reduce chars per line mantra comes from the days when you actually were limited to 80 chars per line. Punch Cards and older hardware was actually limited to 80 chars and back then this made a lot more sense. It actually goes back even further than that, but I digress and am not saying we should hold ourselves to 80 chars. If you would like to read more on this you can find these links amusing and/or enlightening [[1]](https://stackoverflow.com/questions/578059/studies-on-optimal-code-width)[[2]](http://pub.gajendra.net/2012/09/hollerith_tyranny). I do still find there is wisdom on not trying to cram too much on a single line, as it makes it harder to read. That threshold depends on what the project is, for example in the sensu plugin projects we try to limit to no more than 160 chars while in my chef projects I have much larger limits because you deal with deeply nested variable namespaces. Especially reviewing code on systems like github having to scroll horizontally 2-3 times or when it wraps across 3+ lines in an editor. I find that depending on the type of project different line limits should be desired and be enforced, with all rules there will be cases where it reduces the readability by trying to break it up and for that you can always disable the rules inline.

I have been using ruby for now about 4 years and while I certainly disagree on some of the rubocop rules (especially the thresholds) I have found a lot of wisdom in their rule sets overall. I actually had a chat yesterday with another maintainer about this, not line length but in general the rubocop rule sets. Rubocop has made both of us better developers. I know I have learned some very interesting things about some of the deep bowels of ruby from various rubocop rules after digging into the why.